pkg/db/db_dependency.go

```diff
@@ -0,0 +1,88 @@
+package db
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+
+	"github.com/deepfactor-io/javadb/pkg/types"
+	"golang.org/x/xerrors"
+)
+
+const (
+	dependencyDbFileName = "df-java-dependency.db"
+)
+
+func NewDbWithDependency(cacheDir string) (DB, error) {
+	dbPath := filepath.Join(cacheDir, dependencyDbFileName)
+	dbDir := filepath.Dir(dbPath)
+	if err := os.MkdirAll(dbDir, 0700); err != nil {
+		return DB{}, xerrors.Errorf("failed to mkdir: %w", err)
+	}
+
+	// open db
+	var err error
+	db, err := sql.Open("sqlite", dbPath)
+	if err != nil {
+		return DB{}, xerrors.Errorf("can't open db: %w", err)
+	}
+
+	if _, err = db.Exec("PRAGMA foreign_keys=true"); err != nil {
+		return DB{}, xerrors.Errorf("failed to enable 'foreign_keys': %w", err)
+	}
+
+	return DB{
+		client: db,
+		dir:    dbDir,
+	}, nil
+}
+
+func (db *DB) InitDbWithDependency() error {
+	if _, err := db.client.Exec("CREATE TABLE artifacts(id INTEGER PRIMARY KEY, group_id TEXT, artifact_id TEXT)"); err != nil {
+		return xerrors.Errorf("unable to create 'artifacts' table: %w", err)
+	}
+	if _, err := db.client.Exec("CREATE TABLE indices(artifact_id INTEGER, version TEXT, sha1 BLOB, dependency TEXT, foreign key (artifact_id) references artifacts(id))"); err != nil {
+		return xerrors.Errorf("unable to create 'indices' table: %w", err)
+	}
+	if _, err := db.client.Exec("CREATE UNIQUE INDEX artifacts_idx ON artifacts(artifact_id, group_id)"); err != nil {
```

Comment:
I am wondering if we really need a separate table for artifacts? 
Won't a single table with GroupID, ArtifactID, Version, Dependencies be enough? with a unique index on GAV?

Reply:
Wouldnt this increase the size? We'll have an 2 columns with repeated values as compared to a single int primary key

Reply:
Made these changes and i see that the db size has gone up to 4.4GB.

Reply:
Got it. Makes sense.